File: Level 2/피로도/solution.js
function solution(k, dungeons) {
  const visited = Array(dungeons.length).fill(false);
  const answer = dfs(k, 0, dungeons, visited);
  return answer;
}

function dfs(k, count, dungeons, visited) {
  let answer = count;

  dungeons.forEach((dungeon, index) => {
    if (k >= dungeon[0] && !visited[index]) {
      visited[index] = true;
      answer = Math.max(
        answer,
        dfs(k - dungeon[1], count + 1, dungeons, visited)
      );
      visited[index] = false;
    }
  });

  return answer;
}

File: Level 2/피로도/solution.py
def dfs(k, count, dungeons, visited):
    answer = count

    for i in range(len(dungeons)):
        if k >= dungeons[i][0] and not visited[i]:
            visited[i] = True
            answer = max(answer, dfs(k - dungeons[i][1], count + 1, dungeons, visited))
            visited[i] = False

    return answer


def solution(k, dungeons):
    visited = [False] * len(dungeons)
    answer = dfs(k, 0, dungeons, visited)
    return answer
